forms-flow-components/src/components/CustomComponents/InputDropdown.tsx

```diff
@@ -0,0 +1,152 @@
+  }, [dropdownRef]);
+
+  useEffect(() => {
+      
+      if (selectedOption) {
+          setInputValue(selectedOption);
+      }
+  }, [selectedOption]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.value;
+      setInputValue(value);
+
+      const filtered = Options.filter((item) =>
+          item.label.toLowerCase().includes(value.toLowerCase())
+      );
+      setFilteredItems(filtered);
+  };
+
+  const handleSelect = (item: DropdownItem) => {
+      setSelectedItem(item);
+      setInputValue(item.label);
+      setIsDropdownOpen(false);
+      if (item.onClick) {
+          item.onClick(); 
+      }
+  };
+
+  const onFirstItemClick = () => {
+      setTextBoxInput(true);
+      setInputValue('');
+      setSelectedItem(undefined);
+      setIsDropdownOpen(false);
+  };
+
+  const handleClose = () => {
+      setTextBoxInput(false);
+      setInputValue('');
+      setIsDropdownOpen(true);
+  };
+
+  return (
+      <div ref={dropdownRef} className="input-dropdown">
+          {textBoxInput ? (
+              <InputGroup>
+                  <FormInput
+                      value={inputValue}
+                      onChange={(e) => setInputValue(e.target.value)}
+                      aria-label="category input"
+                      icon={<CloseIcon onClick={handleClose} />} 
+                      className="input-with-close"
+                      label={dropdownLabel}
+                  />
+              </InputGroup>
+          ) : (
+              <InputGroup>
+                  <FormInput
+                      placeholder={placeholder}
+                      value={inputValue}
+                      onChange={handleInputChange}
+                      onClick={toggleDropdown}
+                      aria-label="Dropdown input"
+                      icon={<ChevronIcon />}
+                      className={`${isDropdownOpen ? 'border-input collapsed' : ''}`}
+                      onIconClick={toggleDropdown}
+                      label={dropdownLabel}
+                      required={required}
+                  />
+              </InputGroup>
+          )}
+
+          {!textBoxInput && isDropdownOpen && (
+              <ListGroup>
+                  {isAllowInput && (
+                      <ListGroup.Item
+                          onClick={onFirstItemClick}
+                          className="list-first-item-btn"
+                          data-testid="list-first-item"
+                      >
+                          {firstItemLabel}
+                      </ListGroup.Item>
+                  )}
+                  {(filteredItems.length > 0 ? filteredItems : Options).map((item, index) => (
+                      <ListGroup.Item
+                          key={index}
+                          onClick={() => handleSelect(item)}
+                          data-testid={`list-${index}-item`}
+                          aria-label={`list-${index}-item`}
```

Comment:
this should be an english sentence / words

Reply:
Updated.